Move private framework stubs into sparse SDKs

emw-apple opened the pull request:
#### 383e646d32ad677b78f2a28a26ff53b06fe89caf
<pre>
Move private framework stubs into sparse SDKs
<a href="https://bugs.webkit.org/show_bug.cgi?id=259225">https://bugs.webkit.org/show_bug.cgi?id=259225</a>

Reviewed by Alexey Proskuryakov.

Now that we use sparse SDKs for building macOS 14, we can do the same on
other platforms. Move TBDs into the SDKs, at the same paths they exist
in internal SDKs.

* Configurations/SDKAdditions.xcconfig: Fix
  WK_HAS_UNVERSIONED_SDK_ADDITIONS to be YES in open source builds and
  in downlevel macOS builds.

* Configurations/WebKitProjectPaths.xcconfig:
  - Remove definition of WK_PRIVATE_FRAMEWORK_STUBS_DIR.
  - Add additional WK_WORKSPACE_DIR levels for dav1d (nested in libavif,
    which is nested in PAL, which is nested in WebCore!).

In project xcconfigs, remove use of WK_PRIVATE_FRAMEWORK_STUBS_DIR and
WK_DOWNLEVEL_FRAMEWORK_STUBS_DIR.

* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/Scripts/configure-xcode-for-embedded-development:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:

Partially revert 5ddcda389f559c713f179c044018e3e4e2306cfe
(<a href="https://commits.webkit.org/265074@main)">https://commits.webkit.org/265074@main)</a>, which switched a
UseSRCROOTSupportForTAPI environment variable to a user default. Testing
on Xcode 14.3 shows that the defaults value is not being recognized.
This should be safe; my reason for switching it in the first place was
merely to avoid unnecessary environment pollution.

* Tools/Scripts/build-webkit:
* Tools/Scripts/webkitdirs.pm:
(XcodeOptions):

The rest of the patch is SDK creation. Since there are different SDKs
for a platform&apos;s device and simulator variants, but our additions are
the same on both, create symlinks for simulator SDKs. We can replace
these with real SDKs in the future.

* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/FileProvider.framework/FileProvider.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/libFontParser.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppServerSupport.framework/AppServerSupport.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppStoreDaemon.framework/AppStoreDaemon.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/ApplePushService.framework/ApplePushService.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AuthKit.framework/AuthKit.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/BackBoardServices.framework/BackBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/libFontParser.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/FrontBoardServices.framework/FrontBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/InstallCoordination.framework/InstallCoordination.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/VisionKitCore.framework/VisionKitCore.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd: Copied from WebKitLibraries/DownlevelFrameworkStubs/120000/AppleJPEGXL.framework/AppleJPEGXL.tbd.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Copied from WebKitLibraries/WebKitPrivateFrameworkStubs/Mac/120000/libFontParser.tbd.
* WebKitLibraries/SDKs/macosx12.0.internal-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd: Copied from WebKitLibraries/DownlevelFrameworkStubs/120000/AppleJPEGXL.framework/AppleJPEGXL.tbd.
* WebKitLibraries/SDKs/macosx13.0.internal-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist: Remove unnecessary keys.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/Mac/120000/libFontParser.tbd.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/FileProvider.framework/FileProvider.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/libFontParser.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PepperUICore.framework/PepperUICore.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk: Added.

Canonical link: <a href="https://commits.webkit.org/266215@main">https://commits.webkit.org/266215@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e42aa9ec6fa3b76edc116765d64d9a2a976ef48

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13188 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13501 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14926 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12557 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13256 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16009 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15251 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13355 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14027 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11148 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15471 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11315 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11902 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18958 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/11220 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12390 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12069 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15280 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/12479 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12556 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10425 "2 failures") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/13236 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11829 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3476 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3234 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16149 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/13617 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12399 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3271 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist:
```diff
@@ -0,0 +1,22 @@
+        <string>NO</string>
+        <key>FrameworkSearchPaths</key>
+        <array>
+            <string>System/Library/PrivateFrameworks</string>
```
I think I can remove this search path. The projects which link against private frameworks already manually add it via SYSTEM_FRAMEWORK_SEARCH_PATHS.

emw-apple replied:
> I think I can remove this search path. The projects which link against private frameworks already manually add it via SYSTEM_FRAMEWORK_SEARCH_PATHS.

Turns out I cannot do this. I thought the build system would factor out a search path like

    SYSTEM_FRAMEWORK_SEARCH_PATHS = $(SDK_DIR)/System/Library/PrivateFrameworks

into multiple search paths — one for the base SDK and others for each sparse SDK. But it does not.

emw-apple commented on WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist:
```diff
@@ -0,0 +1,22 @@
+        </array>
+        <key>LibrarySearchPaths</key>
+        <array>
+            <string>System/Library/PrivateFrameworks/FontServices.framework</string>
```
Ditto.

emw-apple commented on WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist:
```diff
@@ -0,0 +1,22 @@
+        </array>
+        <key>HeaderSearchPaths</key>
+        <array>
+            <string>usr/local/include</string>
```
Ditto, via
```
SYSTEM_HEADER_SEARCH_PATHS = $(SDK_DIR)$(WK_ALTERNATE_WEBKIT_SDK_PATH)$(WK_LIBRARY_HEADERS_FOLDER_PATH)
```

aproskuryakov commented on Configurations/SDKAdditions.xcconfig:
```diff
@@ -26,10 +26,11 @@
-WK_HAS_VERSIONED_SDK_ADDITIONS_ = $(WK_HAS_VERSIONED_SDK_ADDITIONS$(WK_MACOS_1400));
-WK_HAS_VERSIONED_SDK_ADDITIONS_MACOS_SINCE_1400 = YES;
+WK_HAS_VERSIONED_SDK_ADDITIONS = $(WK_NOT_$(USE_INTERNAL_SDK));
+// Prior to Sonoma, downlevel macOS builds link against AppleJPEGXL.framework.
```
Nitpick: STP too.

aproskuryakov replied:
Less of a nitpick: seems like we will take all open source SDK additions for internal builds, not just the necessary AppleJPEGXL?

emw-apple replied:
We should only take AppleJPEGXL, because that's the only thing in e.g. `macosx13.0.internal-additions.sdk`. The logic to choose between internal and public SDK additions landed in #15464, which I realize makes it difficult to see what's happening from this PR. But in WebKitProjectPaths.xcconfig:

```
WK_VERSIONED_SDK_ADDITIONS_PATH = $(WK_VERSIONED_SDK_ADDITIONS_PATH_$(USE_INTERNAL_SDK));
WK_VERSIONED_SDK_ADDITIONS_PATH_ = $(WK_WORKSPACE_DIR)/WebKitLibraries/SDKs/$(PLATFORM_NAME)$($(DEPLOYMENT_TARGET_SETTING_NAME):base).0-additions.sdk;
WK_VERSIONED_SDK_ADDITIONS_PATH_YES = $(WK_WORKSPACE_DIR)/WebKitLibraries/SDKs/$(PLATFORM_NAME)$($(DEPLOYMENT_TARGET_SETTING_NAME):base).0.internal-additions.sdk;
```

emw-apple committed:
Move private framework stubs into sparse SDKs
https://bugs.webkit.org/show_bug.cgi?id=259225

Reviewed by Alexey Proskuryakov.

Now that we use sparse SDKs for building macOS 14, we can do the same on
other platforms. Move TBDs into the SDKs, at the same paths they exist
in internal SDKs.

* Configurations/SDKAdditions.xcconfig: Fix
  WK_HAS_UNVERSIONED_SDK_ADDITIONS to be YES in open source builds and
  in downlevel macOS builds.

* Configurations/WebKitProjectPaths.xcconfig:
  - Remove definition of WK_PRIVATE_FRAMEWORK_STUBS_DIR.
  - Add additional WK_WORKSPACE_DIR levels for dav1d (nested in libavif,
    which is nested in PAL, which is nested in WebCore!).

In project xcconfigs, remove use of WK_PRIVATE_FRAMEWORK_STUBS_DIR and
WK_DOWNLEVEL_FRAMEWORK_STUBS_DIR.

* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebKit/Configurations/BaseTarget.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/Scripts/configure-xcode-for-embedded-development:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:

Partially revert 5ddcda389f559c713f179c044018e3e4e2306cfe
(https://commits.webkit.org/265074@main), which switched a
UseSRCROOTSupportForTAPI environment variable to a user default. Testing
on Xcode 14.3 shows that the defaults value is not being recognized.
This should be safe; my reason for switching it in the first place was
merely to avoid unnecessary environment pollution.

* Tools/Scripts/build-webkit:
* Tools/Scripts/webkitdirs.pm:
(XcodeOptions):

The rest of the patch is SDK creation. Since there are different SDKs
for a platform's device and simulator variants, but our additions are
the same on both, create symlinks for simulator SDKs. We can replace
these with real SDKs in the future.

* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/FileProvider.framework/FileProvider.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/libFontParser.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/appletvos/16/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppServerSupport.framework/AppServerSupport.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppStoreDaemon.framework/AppStoreDaemon.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/ApplePushService.framework/ApplePushService.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/AuthKit.framework/AuthKit.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/BackBoardServices.framework/BackBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/libFontParser.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/FrontBoardServices.framework/FrontBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/InstallCoordination.framework/InstallCoordination.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/iOS/16/VisionKitCore.framework/VisionKitCore.tbd.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd: Copied from WebKitLibraries/DownlevelFrameworkStubs/120000/AppleJPEGXL.framework/AppleJPEGXL.tbd.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Copied from WebKitLibraries/WebKitPrivateFrameworkStubs/Mac/120000/libFontParser.tbd.
* WebKitLibraries/SDKs/macosx12.0.internal-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd: Copied from WebKitLibraries/DownlevelFrameworkStubs/120000/AppleJPEGXL.framework/AppleJPEGXL.tbd.
* WebKitLibraries/SDKs/macosx13.0.internal-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist: Remove unnecessary keys.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/Mac/120000/libFontParser.tbd.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SDKSettings.plist: Copied from WebKitLibraries/SDKs/macosx14.0-additions.sdk/SDKSettings.plist.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/FileProvider.framework/FileProvider.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/libFontParser.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PepperUICore.framework/PepperUICore.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/URLFormatting.framework/URLFormatting.tbd.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/usr/local/include/AvailabilityProhibitedInternal.h: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk: Added.

Canonical link: https://commits.webkit.org/266215@main
